syntax = "proto3";
package Comunicaciones;

service Comunicador{
    rpc Registro_Vendedor (RegistroVendedor) returns (Status){};
    rpc Registro_Tienda (RegistroTienda) returns (Status){};
    rpc Registro_Asignacion (RegistroAsignacion) returns (Status){};
    rpc listador_tiendas (Empty) returns (stream RegistroTienda){};
    rpc listador_asignaciones (Empty) returns (stream RegistroAsignacion){};
    rpc agregador (stream Producto) returns (Status){};
    rpc listador_productos (Empty) returns (stream Producto){};
}

message Empty {}

message Status{
    optional bool success = 1;
}

message Producto {
    optional int32 id = 1;
    optional int32 cantidad = 2;
    optional string descripcion = 3;
}

message RegistroVendedor{
    optional int32 id = 1;
    optional string nombre = 2;
    optional int32 edad = 3;
    optional double salario = 4;
}

message RegistroTienda{
    optional int32 id = 1;
    optional string descripcion = 2;
    optional string delegacion = 3;
}

message RegistroAsignacion{
    optional int32 idAsignacion = 1;
    optional int32 idTienda = 2;
    optional int32 idVendedor = 3;
}

message ListadoAsignaciones{
    repeated RegistroAsignacion objetos = 1;
}